File: Exercism/fsharp/bob/Bob.fs
module Bob

let (|Silence|_|) (s: string) = if s |> Seq.isEmpty then Some Silence else None
let (|Shouting|_|) (s: string) = if s.ToUpperInvariant() = s && s |> Seq.exists System.Char.IsLetter then Some Shouting else None
let (|Question|_|) (s: string) = if s |> Seq.tryLast = Some '?' then Some Question else None

let response (s: string) =
    match s.Trim() with
      | Question & Shouting -> "Calm down, I know what I'm doing!"
      | Shouting -> "Whoa, chill out!"
      | Question -> "Sure."
      | Silence -> "Fine. Be that way!"
      | _ -> "Whatever."

File: Exercism/csharp/bob/Bob.cs
using System;
using System.Linq;

public static class Bob
{   
    private static bool IsShouting(string s) => s.ToUpperInvariant() == s && s.Any(Char.IsLetter);
    private static bool IsQuestion(string s) => s.Length > 0 && s.Last() == '?';
 
    public static string Response(string statement) => statement.Trim() switch {
      var s when IsQuestion(s) && IsShouting(s) => "Calm down, I know what I'm doing!",
      var s when IsShouting(s) => "Whoa, chill out!",
      var s when IsQuestion(s) => "Sure.",
      var s when s == "" => "Fine. Be that way!",
      _ => "Whatever."
    };
}
